src/skmultiflow/anomaly_detection/half_space_trees.py

```diff
@@ -0,0 +1,557 @@
+from skmultiflow.core import BaseSKMObject, ClassifierMixin
+from skmultiflow.utils.utils import *
+
+
+class HalfSpaceTrees(BaseSKMObject, ClassifierMixin):
+    """
+    Half Space Trees
+
+    Parameters
+    ----------
+
+    n_features: int, required
+        The dimensionality of the stream.
+
+    n_estimators: int, optional (default=25)
+       Number of trees in the ensemble.
+       't' in the original paper.
+
+    window_size: int, optional (default=250)
+        The window size of the stream.
+        'Psi' in the original paper.
+
+    depth: int, optional (default=15)
+        The maximum depth of the trees in the ensemble.
+        'maxDepth' in the original paper.
+
+    size_limit: int, optional (default=50)
+        The minimum mass required in a node (as a fraction of the window size) to calculate the anomaly score.
+        'sizeLimit' in the original paper.
+        A good setting is 0.1 * window_size
+
+    anomaly_threshold: double, optional (default=0.5)
+        The threshold for declaring anomalies.
+        Any instance prediction probability above this threshold will be declared as an anomaly.
+
+    tree_random_state: int (Default: None)
+        Seed for random generation of tree.
+
+    Implements the Streaming Half - Space Trees one - class classifier described in
+    S.C.Tan, K.M.Ting, and T.F.Liu, “Fast anomaly detection for streaming data,”
+    in IJCAI Proceedings - International Joint Conference on Artificial Intelligence,
+    2011, vol. 22, no. 1, pp. 1511–1516.
+    """
+
+    def __init__(self,
+                 n_features,
+                 window_size=250,
+                 depth=15,
+                 n_estimators=25,
+                 size_limit=50,
+                 anomaly_threshold=0.5,
+                 tree_random_state=None):
+
+        super().__init__()
+        self.window_size = window_size
+        self.depth = depth
+        self.n_estimators = n_estimators
+        self.min_values = []
+        self.max_values = []
+        self.ensemble = []
+        self.n_features = n_features
+        self.size_limit = size_limit
+        self.samples_seen = 0
+        self.anomaly_threshold = anomaly_threshold
+        self.is_learning_phase_on = True
+        self.tree_random_state = tree_random_state
+
+    def partial_fit(self, X, y, classes=None, sample_weight=None):
+        """ Partially (incrementally) fit the model.
+
+        Parameters
+        ----------
+        X : numpy.ndarray of shape (n_samples, n_features)
+            The features to train the model.
+
+        y: numpy.ndarray of shape (n_samples)
+            An array-like with the class labels of all samples in X.
+
+        classes: numpy.ndarray, optional (default=None)
+            Array with all possible/known class labels. This is an optional parameter, except
+            for the first partial_fit call where it is compulsory.
+
+        sample_weight: numpy.ndarray of shape (n_samples), optional (default=None)
```

Comment:
This parameter is not used by this method. Please update the documentation to clearly state this fact.

Note: Do not remove the parameter given that it is defined by the `CalssifierMixin` class.

Reply:
Please restore the parameter in the docstring since it is used to generate the documentation. What I mean to say is that it should be stated that the parameter is in the signature but not used, sorry for not being clear.

Suggestion:
```
        classes: None
            Not used by this method.
        sample_weight: None
            Not used by this method.
```